src/libraries/System.Formats.Cbor/src/System/Formats/Cbor/Reader/CborReader.Simple.cs

```diff
@@ -8,7 +8,42 @@ namespace System.Formats.Cbor
 {
     public partial class CborReader
     {
-        private const int SizeOfHalf = 2; // the size in bytes of an IEEE 754 Half-Precision float
+        /// <summary>
+        ///   Reads the next data item as a half-precision floating point number (major type 7).
+        /// </summary>
+        /// <returns>The decoded value.</returns>
+        /// <exception cref="InvalidOperationException">
+        ///   the next data item does not have the correct major type. -or-
+        ///   the next simple value is not a floating-point number encoding. -or-
+        ///   the encoded value is a double-precision float
+        /// </exception>
+        /// <exception cref="CborContentException">
+        ///   the next value has an invalid CBOR encoding. -or-
+        ///   there was an unexpected end of CBOR encoding data. -or-
+        ///   the next value uses a CBOR encoding that is not valid under the current conformance mode.
+        /// </exception>
+        internal Half ReadHalf()
+        {
+            CborInitialByte header = PeekInitialByte(expectedType: CborMajorType.Simple);
+            ReadOnlySpan<byte> buffer = GetRemainingBytes();
+            Half result;
+
+            switch (header.AdditionalInfo)
+            {
+                case CborAdditionalInfo.Additional16BitData:
+                    EnsureReadCapacity(buffer, 1 + HalfHelpers.SizeOfHalf);
+                    result = HalfHelpers.ReadHalfBigEndian(buffer.Slice(1));
+                    AdvanceBuffer(1 + HalfHelpers.SizeOfHalf);
+                    AdvanceDataItemCounters();
+                    return result;
+
+                case CborAdditionalInfo.Additional64BitData:
+                    throw new InvalidOperationException(SR.Cbor_Reader_ReadingAsLowerPrecision);
```

Comment:
Shouldn't Additional32 also hit this message?

Reply:
Fixed, thanks.